virttest/utils_net.py

```diff
@@ -3462,15 +3462,19 @@ def get_ip_address_by_interface(ifname, ip_ver="ipv4", linklocal=False):
     :param ifname: interface name
     :param ip_ver: Host IP version, ipv4 or ipv6.
     :param linklocal: Whether ip address is local or remote
-    :raise NetError: When failed to fetch IP address (ioctl raised IOError.).
+    :raise IPAddrGetError: When failed to fetch IP address.
     """
     if ip_ver == "ipv6":
         ver = netifaces.AF_INET6
         linklocal_prefix = "fe80"
     else:
         ver = netifaces.AF_INET
         linklocal_prefix = "169.254"
-    addr = netifaces.ifaddresses(ifname).get(ver)
+    try:
+        addr = netifaces.ifaddresses(ifname).get(ver)
+    except:
+        raise IPAddrGetError(
+            "Error while retrieving IP address from interface %s." % ifname)
```

Comment:
@yanglei-rh @YongxueHong `IPAddrGetError` would require `mac_addr` as the first argument for its `__init__` method, so this has to be updated.

https://github.com/avocado-framework/avocado-vt/blob/7689bcddf78eb5282a5980cd697af6e4ebbbd298/virttest/utils_net.py#L307-L309

Reply:
And also, the error message indicated that it was for guest issues, therefore I'm not sure if it would make sense since our case is a host one, what do you think?

https://github.com/avocado-framework/avocado-vt/blob/7689bcddf78eb5282a5980cd697af6e4ebbbd298/virttest/utils_net.py#L315-L316

Reply:
Hi @luckyh 
Yeah, I forgot the usage of this Error.

Reply:
I re-checked the subclass of `NetError`, there is no other available error for this function.
Maybe we keep the original type.
Sorry for disturbing you.

Reply:
Hello @luckyh @PaulYuuu Based the above comment, it should be changed `NetError`, right?